server/issue.go

```diff
@@ -357,6 +342,14 @@ func (s *Server) createPipelineFromIssue(ctx context.Context, issueCreate *api.I
 		return nil, echo.NewHTTPError(http.StatusBadRequest, err.Error()).SetInternal(err)
 	}
 
+	ok, err := s.canPrincipalBeAssignee(ctx, creatorID, environmentID, issueCreate.ProjectID, issueCreate.Type)
```

Comment:
We need a better comment on this explaining why doing this check. e.g. Why we check creatorId instead of issueCreate.AssigneeID ?